xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -144,6 +152,41 @@ protected boolean delete(XWikiContext context) throws XWikiException
         }
     }
 
+    private boolean deleteAllFromRecycleBin(XWikiContext context) throws XWikiException
+    {
+        RefactoringScriptService refactoring =
+            (RefactoringScriptService) Utils.getComponent(ScriptService.class, "refactoring");
+        PermanentlyDeleteRequest deleteRequest = refactoring.createPermanentlyDeleteRequest(Collections.emptyList());
```

Comment:
I don't see how this can work for a subwiki since you never indicate the wiki for which you want to empty the trash bin (and I doubt your goal is to empty the trash bin of all wiki, that's definitely not what the job is doing).

Reply:
Indeed I could not see it, and I just found initializeContext on job side. So it should be OK.